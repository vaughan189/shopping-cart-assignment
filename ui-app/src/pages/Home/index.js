/* eslint-disable react-hooks/exhaustive-deps */
import { Box, Container, Grid } from "@mui/material";
import React, { useEffect } from "react";
import useAxios from "../../api/useAxios";
import { Carousal } from "../../components/Carousel";
import { Categories } from "../../components/Categories";

const Home = () => {
  const {
    response: bannerData,
    loading: bannerDataLoader,
    // error: bannerDataError,
    fetchData: fetchBannerData,
  } = useAxios();

  const {
    response: categoriesData,
    loading: categoriesDataLoader,
    // error: categoriesDataError,
    fetchData: fetchCategories,
  } = useAxios();

  const getBanners = () => {
    fetchBannerData({
      method: "GET",
      url: "http://localhost:3000/banners",
      headers: {
        accept: "*/*",
      },
    });
  };

  const getCategories = () => {
    fetchCategories({
      method: "GET",
      url: "http://localhost:3000/categories",
      headers: {
        accept: "*/*",
      },
    });
  };

  useEffect(() => {
    getBanners();
    getCategories();
  }, []);

  return (
    <Box component="main" sx={{ marginTop: "5%" }}>
      <Container maxWidth="xl">
        <Grid container direction="column" spacing={2}>
          {!bannerDataLoader && !categoriesDataLoader && (
            <>
              <Grid container direction="row">
                <Grid item xs={12} sm={12} md={12} xl={12}>
                  <Carousal bannerData={bannerData.data} />
                </Grid>
                <Grid item xs={12} sm={12} md={12} xl={12}>
                  <Categories categoriesData={categoriesData.data} />
                </Grid>
              </Grid>
            </>
          )}
        </Grid>
      </Container>
    </Box>
  );
};

export default Home;
