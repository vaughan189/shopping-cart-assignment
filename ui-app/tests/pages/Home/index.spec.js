import React from "react";
import { shallow } from "enzyme";
import Home from "../../../src/pages/Home";

// jest.mock("../../../src/api/useAxios.js", () => ({
//   useAxios: () => ({
//     response: [],
//     loading: false,
//     fetchData: () => {},
//   }),
// }));

describe("Home Page", () => {
  it("should render the Home Page correctly", () => {
    const component = shallow(<Home debug />);
    expect(component).toMatchSnapshot();
  });
});
