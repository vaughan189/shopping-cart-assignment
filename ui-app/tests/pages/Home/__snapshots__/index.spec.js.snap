// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page should render the Home Page correctly if loading is false 1`] = `
<ForwardRef(Box)
  component="main"
  sx={
    Object {
      "marginTop": "5%",
    }
  }
>
  <ForwardRef(Container)
    maxWidth="xl"
  >
    <ForwardRef(Grid)
      container={true}
      direction="column"
      spacing={2}
    >
      <ForwardRef(Grid)
        container={true}
        direction="row"
      >
        <ForwardRef(Grid)
          item={true}
          md={12}
          sm={12}
          xl={12}
          xs={12}
        >
          <Carousal />
        </ForwardRef(Grid)>
        <ForwardRef(Grid)
          item={true}
          md={12}
          sm={12}
          xl={12}
          xs={12}
        >
          <Categories />
        </ForwardRef(Grid)>
      </ForwardRef(Grid)>
    </ForwardRef(Grid)>
  </ForwardRef(Container)>
</ForwardRef(Box)>
`;

exports[`Home Page should render the Home Page correctly if loading is true 1`] = `
<ForwardRef(Box)
  component="main"
  sx={
    Object {
      "marginTop": "5%",
    }
  }
>
  <ForwardRef(Container)
    maxWidth="xl"
  >
    <ForwardRef(Grid)
      container={true}
      direction="column"
      spacing={2}
    />
  </ForwardRef(Container)>
</ForwardRef(Box)>
`;
